src/rantanplan/structures.py

```diff
@@ -0,0 +1,659 @@
+import re
+import statistics
+from collections import Counter
+
+# Stanza structures where each tuple is defined as follows:
+# (
+#     CONSONANT_RHYME | ASSONANT_RHYME,
+#     "structure name",
+#     r".*",  # regular expression to match the rhymed line pattern
+#     lambda lengths: True  # function checking a condition on line lengths
+# )
+# Structures will be checked in order of definition, the first one to match
+# will be chosen.
+
+
+def most_common_lengths(lengths_list, n=4, number_of_verses=None):
+    """Return the n most commons lengths in a list of verses lengths
+
+    :param lengths_list: List with the lengths to be checked
+    :param n: Number of most common lengths to be returned
+    :param number_of_verses: How many verses are in a given stanza
+    :return: The n most common lengths
+    """
+    most_common_lengths_list = []
+    count = Counter(lengths_list).most_common(n)
+    if len(lengths_list) == number_of_verses or number_of_verses is None:
+        for length in count:
+            # `lengths` is a tuple with the length and it's count so we only get
+            # the first element of the tuple
+            most_common_lengths_list.append(length[0])
+    return most_common_lengths_list
+
+
+def is_seguidilla_compuesta(lengths):
+    """
+    [7, 5, 7, 5, 5, 7, 5]
+    :param lengths:
+    :return:
+    """
+    correct_lines = 0
+    number_of_verses = len(lengths)
+    if number_of_verses == 7:
+        for i in (0, 2, 5):
+            if 9 > lengths[i] > 5:
+                correct_lines += 1
+        for i in (1, 3, 4, 6):
+            if 7 > lengths[i] > 3:
+                correct_lines += 1
+    return correct_lines == number_of_verses
+
+
+def is_seguidilla(lengths):
+    """
+    [7, 5, 7, 5]
+    :param lengths:
+    :return:
+    """
+    correct_lines = 0
+    number_of_verses = len(lengths)
+    if number_of_verses == 4:
+        for i in (0, 2):
+            if 9 > lengths[i] > 5:
+                correct_lines += 1
+        for i in (1, 3):
+            if 7 > lengths[i] > 3:
+                correct_lines += 1
+    if correct_lines == number_of_verses:
+        return True
+    return False
```

Comment:
Replace with `return correct_lines == number_of_verses` in all functions where applicable.